### PR TITLE
feat(Carousel): add RTL keyboard support

### PR DESCRIPTION
**Pull request by kyletsang**
Part of #5753

### CODE REVIEW

File: src/ThemeProvider.tsx [hunk 1]
```diff
@@ -2,31 +2,51 @@ import PropTypes from 'prop-types';
...
-const ThemeContext = React.createContext({});
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface ThemeProviderProps extends Partial<ThemeContextValue> {}
+
+const ThemeContext = React.createContext<ThemeContextValue>({ prefixes: {} });
 const { Consumer, Provider } = ThemeContext;
 
-function ThemeProvider({ prefixes, children }) {
-  const copiedPrefixes = useMemo(() => ({ ...prefixes }), [prefixes]);
+const ThemeProvider: React.FC<Partial<ThemeContextValue>> = ({
```
Comment (jquense) [inline src/ThemeProvider.tsx:16]: what's the reason btw for switching to an arrow function here? I find the typing slightly harder to understand vs a plain function

File: src/ThemeProvider.tsx [hunk 2]
```diff
@@ -2,31 +2,51 @@ import PropTypes from 'prop-types';
...
 export function useBootstrapPrefix(
   prefix: string | undefined,
   defaultPrefix: string,
 ): string {
-  const prefixes = useContext(ThemeContext);
+  const { prefixes } = useContext(ThemeContext);
   return prefix || prefixes[defaultPrefix] || defaultPrefix;
 }
 
+export function useRTL() {
```
Comment (jquense) [inline src/ThemeProvider.tsx:45]:
```suggestion
export function useIsRTL() {
```
Maybe? just a thought

File: src/ThemeProvider.tsx [hunk 3]
```diff
@@ -2,31 +2,51 @@ import PropTypes from 'prop-types';
...
+  const { dir } = useContext(ThemeContext);
+  return dir === 'rtl';
```
Comment (jquense) [inline src/ThemeProvider.tsx:47]: What alternatives to storing this in context have we considered? This adds a new RB specific api required in addition to the upstream one. What about stealing their JS helper? https://github.com/twbs/bootstrap/blob/main/js/src/util/index.js#L230 

If we want a reactive hook still could combine with a mutationObserver to trigger updates, though i'm not sure how common dynamically changing the read direction is aside from demonstrations in docs

File: src/ThemeProvider.tsx (hunk 2 above)
Comment (kyletsang) [reply src/ThemeProvider.tsx:45]: Yeah `useIsRTL` is more clear.  I flip flopped between the 2 when working on it 😛

File: src/ThemeProvider.tsx (hunk 3 above)
Comment (kyletsang) [reply src/ThemeProvider.tsx:47]: I used the context to support SSR since `document` wasn't available.  Wasn't sure if there was another way?

Comment (jquense) [reply src/ThemeProvider.tsx:47]: hmm you are right. what if we made this a function? presumably this is actually only relevant on the client, but we'd need to defer accesing the document til after render...or maybe set `dir` state in a useEffect? Do we think there is a reason why we'd need this during a SSR run, i suspect most usage is in event handlers.

Comment (kyletsang) [reply src/ThemeProvider.tsx:47]: From my list before, the components that need RTL changes are dropdown, modal, carousel, tooltip.  I think you're right that these aren't really needed during SSR - will have to double check.

Comment (kyletsang) [reply src/ThemeProvider.tsx:47]: @jquense looking at upstream, this is where RTL is used:
- Modal - setting padding for scrollbar when modal is shown
- Dropdown - menu placement
- Tooltip/Popover - tooltip/popover placement
- Carousel - keyboard control

Seems like we would need SSR support for dropdowns with renderOnMount or tooltips/popovers with show=true?

**Review by jquense — APPROVED** [review]

**Pull request merged**